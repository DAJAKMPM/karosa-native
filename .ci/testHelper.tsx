import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
