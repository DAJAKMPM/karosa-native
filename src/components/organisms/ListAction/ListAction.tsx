/**
 *
 * ListAction
 * @format
 *
 */

import React from "react";
import { View, Image, TouchableOpacity } from "react-native";

// import { BaseText } from "../../base-text";
import Text from "@app/atoms/Text";

import { Props } from "./types";
import { styles } from "./styles";

const ListAction: React.FC<Props> = ({
  text,
  image,
  icon,
  style,
  action,
  onClick,
  structure,
}) => {
  return (
    <React.Fragment>
      {onClick ? (
        <TouchableOpacity onPress={onClick}>
          <View style={[styles.container, style?.containerStyle]}>
            <View style={styles.leftContainer}>{structure?.LeftComponent}</View>
            {structure?.RightComponent}
          </View>
        </TouchableOpacity>
      ) : (
        <View style={[styles.container, style?.containerStyle]}>
          {icon?.left}
          {image && <Image style={styles.image} source={image} />}
          <View style={styles.leftContainer}>
            {/* <BaseText
              customStyles={
                image ? styles.hasImage : [styles.left, style?.textStyle?.left]
              }
              numberOfLines={1}>
              {text?.left}
            </BaseText> */}
            <Text
              text={text?.left || ""}
              numberOfLines={1}
              textStyle={
                image ? styles.hasImage : [styles.left, style?.textStyle?.left]
              }
            />
          </View>

          {action && (
            <TouchableOpacity onPress={action}>
              {text?.right && (
                // <BaseText style={[styles.right, style?.textStyle?.right]}>
                //   {text?.right}
                // </BaseText>
                <Text
                  text={text?.right}
                  textStyle={[styles.right, style?.textStyle?.right]}
                />
              )}
              {icon?.right}
            </TouchableOpacity>
          )}

          {structure?.RightComponent}
        </View>
      )}
    </React.Fragment>
  );
};

export default ListAction;
