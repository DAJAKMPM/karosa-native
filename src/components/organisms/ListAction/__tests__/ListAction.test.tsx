import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import ListAction from "../ListAction";

describe("ListAction Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<ListAction />);

    expect(toJSON()).toMatchSnapshot();
  });
});
