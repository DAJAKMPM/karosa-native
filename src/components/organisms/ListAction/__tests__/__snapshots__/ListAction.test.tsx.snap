// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListAction Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 13,
          "paddingLeft": 1,
          "paddingRight": 13,
          "paddingTop": 13,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginLeft": 15,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#1C1C1C",
            "fontFamily": "SFProText-Regular",
            "fontSize": 16,
            "fontWeight": "500",
            "lineHeight": 18.75,
          }
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
